frontend/src/lib-components/employee/messages/MessageEditor.tsx

```diff
@@ -217,39 +199,20 @@ export default React.memo(function MessageEditor({
   useEffect(
     function syncDraftContentOnMessageChanges() {
       contentTouched &&
-        setDraft(
-          messageToUpdatableDraftWithAccount(message, selectedReceiverOptions)
-        )
+        setDraft(messageToUpdatableDraftWithAccount(message, selectedReceivers))
     },
-    [contentTouched, message, selectedReceiverOptions, setDraft]
-  )
-
-  const updateReceiverTree = useCallback(
-    (newSelection: ReactSelectOption[]) => {
-      setReceiverTree((old) =>
-        createReceiverTree(
-          old,
-          newSelection.map((s) => s.value)
-        )
-      )
-    },
-    []
+    [contentTouched, message, selectedReceivers, setDraft]
   )
 
   useEffect(
     function updateSelectedReceiversOnReceiverTreeChanges() {
-      if (!receiverTree) {
-        return
-      }
-
-      const selected = getSelected(receiverTree)
       setMessage((old) => ({
         ...old,
-        recipientIds: selected.map((s) => s.selectorId),
-        recipientNames: selected.map((s) => s.name)
+        recipientIds: selectedReceivers.map((s) => s.key),
+        recipientNames: selectedReceivers.map((s) => s.text)
       }))
     },
-    [receiverTree]
+    [selectedReceivers]
```

Comment:
Both `useEffect`s here should be removed as per [You Might Not Need an Effect](https://beta.reactjs.org/learn/you-might-not-need-an-effect).

When `useEffect` is used to adjust state based on user input (message content or recipient changes in these cases), it would be better to do the state update in the event handler itself. Using `useEffect` causes unnecessary rerenders.